Obsidian.API/_Interfaces/IConfig.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace Obsidian.API;
 
-public interface IConfig
+public interface IServerConfiguration
```

Comment:
Any specific reason to rename this interface?

Reply:
Mostly to avoid confusion. The reason I did not want it to be named something like `IConfig` or `IConfiguration` is because a generic host will automatically add a `Microsoft.Extensions.IConfiguration` to the DI container. So to avoid confusion, I made it clear this is a server configuration, not an application configuration.